### PR TITLE
Pass the full image and roi rectangle only

### PR DESCRIPTION
**Pull request by kingst**

### CODE REVIEW

File: CardScan/Classes/ScanBaseViewController.swift [hunk 1]
```diff
@@ -317,8 +317,18 @@ public protocol TestingImageDataSource: AnyObject {
         return squareCardImage.cropping(to: cardRect) ?? squareCardImage
     }
     
+    func toSquareCardImage(fullCardImage: CGImage, roiRectangle: CGRect) -> CGImage? {
+        let width = CGFloat(fullCardImage.width)
+        let height = width
+        let centerY = (roiRectangle.maxY + roiRectangle.minY) * 0.5
```
[the remark this answers is not in the document]
Comment (kingst) [reply CardScan/Classes/ScanBaseViewController.swift]: No reason in particular, do you think using `minY + height * 0.5` is more intuitive?

File: CardScan/Classes/ScanBaseViewController.swift [hunk 2]
```diff
@@ -317,8 +317,18 @@ public protocol TestingImageDataSource: AnyObject {
...
+        let cropRectangle = CGRect(x: 0.0, y: centerY - height * 0.5,
+                                   width: width, height: height)
+        return fullCardImage.cropping(to: cropRectangle)
+    }
+    
     @available(iOS 11.2, *)
-    open func blockingMlModel(squareCardImage: CGImage, fullCardImage: CGImage) {
+    open func blockingMlModel(fullCardImage: CGImage, roiRectangle: CGRect) {
+        guard let squareCardImage = toSquareCardImage(fullCardImage: fullCardImage, roiRectangle: roiRectangle) else { return }
```
[the remark this answers is not in the document]
Comment (kingst) [reply CardScan/Classes/ScanBaseViewController.swift]: yes, exactly. I wanted to simplify this down to its basics

File: CardScan/Classes/ScanBaseViewController.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaimejiyepark) [reply CardScan/Classes/ScanBaseViewController.swift]: no just a thought :-)

File: CardScan/Classes/ScanBaseViewController.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaimejiyepark) [reply CardScan/Classes/ScanBaseViewController.swift]: sweeet!

**Comment by jaimejiyepark** [comment]
lgtm! #shipit

**Pull request merged**